extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -1818,8 +1831,19 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
     [SERVICE_TYPES.MIGRATE_ACCOUNTS]: migrateAccounts,
     [SERVICE_TYPES.ADD_ASSETS_LIST]: addAssetsList,
     [SERVICE_TYPES.MODIFY_ASSETS_LIST]: modifyAssetsList,
+    [SERVICE_TYPES.GET_IS_ACCOUNT_MISMATCH]: getIsAccountMismatch,
   };
 
+  const currentState = sessionStore.getState();
+  const publicKey = publicKeySelector(currentState);
+
+  if (
+    request.activePublicKey &&
+    request.activePublicKey !== publicKey &&
+    request.type !== SERVICE_TYPES.GET_IS_ACCOUNT_MISMATCH
+  ) {
+    return { error: "Public key does not match active public key" };
```

Comment:
user will return early here in case of account mismatch; the request will not make it through to the messageResponder

Reply:
nice safeguard